hub-stack-elaborate

```diff
@@ -66,81 +57,131 @@ while [ "$1" != "" ]; do
     shift
 done
 
-echo "Reconciling deployment plan for $HUB_STACK_NAME"
-echo "* Using domain name: $HUB_DOMAIN_NAME"
 
-if test -z "$HUB_FILES"; then
-  cat << EOF
-Error: cannot find hubctl definition files (HUB_FILES)"
+is_url() {
+  # shellcheck disable=SC2317
+  echo "$1" | grep -e '[[:alnum:]]*://' >/dev/null 2>&1
+}
 
-Probably because stack has not been configured for deployment yet!
+is_abs() {
+  case "$1" in
+    /*) return 0 ;;
+    *) return 1 ;;
+  esac
+}
 
-Example:
-  $ hubctl configure -f hub.yaml
+find_first() {
+  result=
+  while test "$1" != "" -a -z "$result"; do
+    result="$(files find-in-path "$1")"
+    shift
+  done
+  echo "$result"
+}
 
-EOF
+abs_files() {
+  while read -r f; do
+    if is_url "$f"; then
+      echo "$f"
+    elif test -f "$f"; then
+      files abspath "$f"
+    elif test -f "$HUB_WORKDIR/$f"; then
+      files abspath "$HUB_WORKDIR/$f"
+    elif ! is_abs "$f"; then
+      files abspath "$HUB_WORKDIR/$f"
+    else
+      echo "$f"
+    fi
+  done
+  set +x
+}
 
-  exit 2
-fi
+rel_files() {
+  while [ "$1" != "" ]; do
+    if test -f "$1"; then
+      printf '%s ' "$(files relpath "$1" "$HUB_WORKDIR")"
+    elif test -f "$HUB_WORKDIR/$1"; then
+      printf '%s ' "$(files relpath "$HUB_WORKDIR/$1" "$HUB_WORKDIR")"
+    else
+      color e "Error: file not found: $1"
+      exit 1
+    fi
+    shift
+  done
+}
 
-if test -z "$HUB_STATE"; then
-  if test -f ".hub/$HUB_DOMAIN_NAME.state"; then
-    HUB_STATE=".hub/$HUB_DOMAIN_NAME.state"
-  fi
-fi
+split() {
+  sed "s/$1/\\n/g"
+}
 
-# if test -n "$HUB_STATE"; then
-#   echo "* Using hubctl state: $HUB_STATE"
-#   HUB_OPTS="$HUB_OPTS -s $HUB_STATE"
-# fi
+join() {
+  xargs | sed "s/ /$1/g"
+}
 
-if test -z "$HUB_ELABORATE"; then
-  HUB_ELABORATE=".hub/$HUB_DOMAIN_NAME.elaborate"
-fi
+printf "Elaborating stack: "; color h "$HUB_STACK_NAME"
 
-cat << EOF
+# shellcheck disable=SC2086
 
-Proceeding with: elaborate
-* Result will be saved saved to: $HUB_ELABORATE
-* Running CLI command:
+hub_files="$(echo "$HUB_FILES" | split " " | abs_files | join " ")"
 
-  $ hubctl elaborate $HUB_FILES -o $HUB_ELABORATE $HUB_OPTS
+if test -z "$hub_files"; then
+  color err <<END
+Error: cannot find hubctl definition files (HUB_FILES)"
 
-EOF
+Probably because stack has not been configured for deployment yet!
 
-verb="elaborate"
-work_dir="$(pwd)"
-script_dir=$(dirname "$0")
+Example:
+  $ hubctl configure -f hub.yaml
```

Comment:
Probably it should be `hubctl stack init`

Reply:
Indeed this looks like an code dated to "before" we have got `hub-stack-init`